java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/GetterSetterGenerator.java

```diff
@@ -18,58 +18,175 @@
+import java.util.stream.Collectors;
 import javax.lang.model.element.Modifier;
 import javax.lang.model.element.TypeElement;
 import javax.lang.model.element.VariableElement;
 import javax.lang.model.util.ElementFilter;
 import org.eclipse.lsp4j.ApplyWorkspaceEditParams;
-import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.CodeAction;
+import org.eclipse.lsp4j.CodeActionKind;
+import org.eclipse.lsp4j.CodeActionParams;
+import org.eclipse.lsp4j.MessageParams;
+import org.eclipse.lsp4j.MessageType;
 import org.eclipse.lsp4j.Range;
 import org.eclipse.lsp4j.TextEdit;
 import org.eclipse.lsp4j.WorkspaceEdit;
-import org.eclipse.lsp4j.services.LanguageClient;
 import org.netbeans.api.java.source.CodeStyle;
 import org.netbeans.api.java.source.CompilationInfo;
 import org.netbeans.api.java.source.ElementUtilities;
 import org.netbeans.api.java.source.JavaSource;
 import org.netbeans.api.java.source.TreeUtilities;
 import org.netbeans.modules.java.editor.codegen.GeneratorUtils;
-import static org.netbeans.modules.java.lsp.server.protocol.TextDocumentServiceImpl.fromUri;
+import org.netbeans.modules.java.lsp.server.Utils;
 import org.openide.filesystems.FileObject;
+import org.openide.util.NbBundle;
 import org.openide.util.Pair;
 
 /**
  *
  * @author lahvac
  */
-public class GetterSetterGenerator {
+public final class GetterSetterGenerator extends CodeGenerator {
 
-    private static final String ERROR = "<error>"; //NOI18N
+    public static final String GENERATE_GETTERS =  "java.generate.getters";
+    public static final String GENERATE_SETTERS =  "java.generate.setters";
+    public static final String GENERATE_GETTERS_SETTERS =  "java.generate.getters.setters";
 
-    public static Pair<Set<VariableElement>, Set<VariableElement>> findMissingGettersSetters(CompilationInfo info, Range range, boolean all) {
-        TreePath tp = info.getTreeUtilities().pathFor(getOffset(info, range.getStart()));
+    private final Set<String> commands = Collections.unmodifiableSet(new HashSet(Arrays.asList(GENERATE_GETTERS, GENERATE_SETTERS, GENERATE_GETTERS_SETTERS)));
+    private final Gson gson = new Gson();
+
+    GetterSetterGenerator() {
+    }
 
-        while (tp != null && !TreeUtilities.CLASS_TREE_KINDS.contains(tp.getLeaf().getKind())) {
-            tp = tp.getParentPath();
+    @Override
+    @NbBundle.Messages({
+        "DN_GenerateGetters=Generate Getters...",
+        "DN_GenerateSetters=Generate Setters...",
+        "DN_GenerateGettersSetters=Generate Getters and Setters...",
+        "DN_GenerateGetterFor=Generate Getter for \"{0}\"",
+        "DN_GenerateSetterFor=Generate Setter for \"{0}\"",
+        "DN_GenerateGetterSetterFor=Generate Getter and Setter for \"{0}\"",
+    })
+    public List<CodeAction> getCodeActions(CompilationInfo info, CodeActionParams params) {
+        List<String> only = params.getContext().getOnly();
+        boolean all = only != null && only.contains(CodeActionKind.Source);
```

Comment:
The other generators bailed out if only != null, but did not contain `CodeActionKind.Source`; this seem to process all eligible fields later in `processCommand`. Right now just `QuickFix` is declared (in addition to `Source`), but better check explicitly so it own't break when the set of `CodeActionKind`s expand later.